### DISCUSSION TITLE
Variables with quarto_render

### DISCUSSION THREAD

**Discussion by chefdoctorsam** [quarto-dev/quarto-cli#1787] [Q&A]
I have been unsuccessful trying to get quarto_render to work similarly to rmarkdown::render. I want to generate variables/objects in one Quarto document and run quarto_render from that document to put them into a report generated by another Quarto document. I probably didn't use any of the right words to describe this, but it's exactly the example on this page: https://bookdown.org/yihui/rmarkdown-cookbook/rmarkdown-render.html

Every time I try I get the error `Error in eval(parse_only(code), envir = envir) : object 'test_value' not found
Calls: .main ... inline_exec -> hook_eval -> withVisible -> eval -> eval
Execution halted`

This is the document where the variables live, called test.qmd:
````
---
title: "test"
format: html
editor: visual
---

```{r}
library(quarto)
test_value <- 111
quarto_render(input = "test_report.qmd")
```
````

This is the document where the report lives, called test_report.qmd.
````
---
title: "test_report"
format: html
editor: visual
---

```{r}
test_value
```
````

The code above runs successfully if quarto_render is changed to rmarkdown::render. I can't find anything indicating that this functionality is supposed to work differently in Quarto, but I could be searching for the wrong thing.

**Comment by mcanouil** [comment]
The URL is not properly formatted in your message.
Could you put the code you are using directly in here? (It will make things way easier/quicker 😉)
Following the guidelines <https://yihui.org/issue/#please-format-your-issue-correctly>

**Reply by chefdoctorsam** [reply]
@mcanouil Fixed the link and put a minimal example in my original post. Thanks for your help!

**Reply by mcanouil** [reply]
short answer, I would use parameterised report, because here environnement is not shared, nor passed.
Quarto can use R but it is not integrated in R (or Python, Julia, etc.) contrary to knitr/rmarkdown.

(I did not have time to test the following, but it should work if I did not make any typos)

````
---
title: "test"
format: html
editor: visual
---

```{r}
library(quarto)
test_value <- 111
quarto_render(input = "test_report.qmd", execute_params = list(test_value = test_value))
```
````


````
---
title: "test_report"
format: html
editor: visual
params:
  test_value: 0
---

```{r}
params$test_value
```
````

**Reply by chefdoctorsam** [reply]
Thank you for this explanation. Does this mean that there isn't a way to pass something like a ggplot object or a gt table through?

**Reply by mcanouil** [reply]
I think it should work, but one has to try to be sure.

**Reply by adpadi** [reply]
I think you're asking [the same question I asked a couple days ago](https://github.com/quarto-dev/quarto-cli/discussions/1747)

You'll need to add a param line to the YAML. Then you can access it as a list. 

```
---
title: "My Document"
params:
  alpha: 0.1
  ratio: 0.1
---
```

Then call it using e.g. ` params$alpha`

In your quarto_render use the `execute_params=list(arg=x)` parameter

**Reply by mcanouil** [reply]
Partly related but not the same question.
Also I already answered 😉
